### PR TITLE
Refactor project, tap etc.

### PR DESCRIPTION
**Pull request by Uzlopak**
This will streamline the repo.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,8 +31,9 @@
     "eslint-plugin-prettier": "^4.0.0",
     "fastify": "^4.2.0",
     "fp-ts": "^2.11.2",
-    "jest": "^29.0.1",
     "prettier": "^2.4.1",
```
Comment (Eomm) [inline package.json:34]: We could remove it as well with `eslint-plugin-prettier`

File: index.js [hunk 2]
```diff
@@ -1,15 +1,64 @@
...
+
+  next()
+}
+
+function isEither (payload) {
+  return payload.left || payload.right
+}
+
+function isPromise (value) {
+  return utilTypes.isPromise(value)
```
Comment (Eomm) [inline index.js:54]: OMG!

File: index.js [hunk 3]
```diff
@@ -1,15 +1,64 @@
...
+          .then((taskResult) => {
+            if (isEither(taskResult)) {
+              return resolvePayload(done, taskResult.left, taskResult.right, res)
+            }
+
+            return resolvePayload(done, null, taskResult, res)
+          })
+          .catch((err) => {
+            return done(err)
+          })
```
Comment (Eomm) [inline index.js:34]:
```suggestion
          .catch(done)
```

File: test/funkyPlugin.test.js [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+    return Promise.resolve(either.right(DUMMY_USER))
+  }
+
+  const app = await initAppGet(t, route).ready()
+
+  t.teardown(() => {
+    app.close()
+  })
+
+  await assertCorrectResponse(t, app)
```
Comment (Eomm) [inline test/funkyPlugin.test.js:34]: not a fan about this helper, but it was already there 🙌🏽

Comment (Uzlopak) [reply test/funkyPlugin.test.js:34]: I just wanted to keep as much structure as possible the same to not make it impossible to review it without having to understand the whole plugin

File: package.json (hunk 1 above)
Comment (Uzlopak) [reply package.json]: Yeah missed to remove the packages.

File: index.js (hunk 2 above)
Comment (Uzlopak) [reply index.js:54]: That is wrapped or that it is using node native isPromise?

Comment (Eomm) [reply index.js:52]: I did not know this utility :D

**Review by Eomm — APPROVED** [review]

**Pull request merged**